cdap-examples/WikipediaPipeline/src/main/java/co/cask/cdap/examples/wikipedia/WikipediaService.java

```diff
@@ -73,6 +67,9 @@ protected void configure() {
     @Path("/lda/topics")
     public void getTopics(HttpServiceRequest request, HttpServiceResponder responder) {
       List<Integer> topics = new ArrayList<>();
+      String dataNamespace = getContext().getRuntimeArguments().get("namespace");
+      dataNamespace = dataNamespace == null ? getContext().getNamespace() : dataNamespace;
+      clusteringTable = getContext().getDataset(dataNamespace, WikipediaPipelineApp.SPARK_CLUSTERING_OUTPUT_DATASET);
```

Comment:
As these lines are repeated numerous times (lines 70 - 72), what about putting them into a utility function, passing in the dataset name?